Loop/Managers/TrustedTimeChecker.swift

```diff
@@ -0,0 +1,75 @@
+        set {
+            set(newValue, forKey: Key.lastSignificantTimeChangeAlert.rawValue)
+        }
+    }
+}
+
+class TrustedTimeChecker {
+    private let acceptableTimeDelta = TimeInterval.seconds(120)
+    private let minimumAlertFrequency = TimeInterval.minutes(30)
+
+    // For NTP time checking
+    private var ntpClient: TrueTimeClient
+    private weak var alertManager: AlertManager?
+    private lazy var log = DiagnosticLog(category: "TrustedTimeChecker")
+
+    init(_ alertManager: AlertManager) {
+        ntpClient = TrueTimeClient.sharedInstance
+        ntpClient.start()
+        self.alertManager = alertManager
+        NotificationCenter.default.addObserver(forName: UIApplication.significantTimeChangeNotification,
+                                               object: nil, queue: nil) { [weak self] _ in self?.checkTrustedTime() }
+        NotificationCenter.default.addObserver(forName: .LoopRunning,
+                                               object: nil, queue: nil) { [weak self] _ in self?.checkTrustedTime() }
+    }
+    
+    private func checkTrustedTime() {
+        ntpClient.fetchIfNeeded { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .success(referenceTime):
+                let deviceNow = Date()
+                let ntpNow = referenceTime.now()
+                let timeDelta = abs(ntpNow.timeIntervalSince(deviceNow))
+                let timeSinceLastAlert = abs(ntpNow.timeIntervalSince(UserDefaults.standard.lastSignificantTimeChangeAlert ?? Date.distantPast))
+                if timeDelta > self.acceptableTimeDelta, timeSinceLastAlert > self.minimumAlertFrequency {
+                    self.log.info("applicationSignificantTimeChange: ntpNow = %@, deviceNow = %@", ntpNow.debugDescription, deviceNow.debugDescription)
+                    self.issueTimeChangedAlert()
+                    UserDefaults.standard.lastSignificantTimeChangeAlert = ntpNow
+                }
+            case let .failure(error):
+                self.log.error("applicationSignificantTimeChange: Error getting NTP time: %@", error.localizedDescription)
+            }
+        }
+    }
+
+    private func issueTimeChangedAlert() {
+        let alertIdentifier = Alert.Identifier(managerIdentifier: "Loop", alertIdentifier: "significantTimeChange")
+        let alertTitle = NSLocalizedString("Time Change Detected", comment: "Time change alert title")
+        // TODO: remove Tidepool-isms
+        let alertBody = NSLocalizedString("Your phone’s time has been changed. Tidepool Loop needs accurate time records to make predictions about your glucose and adjust your insulin accordingly.\n\nCheck in your iPhone Settings (General / Date & Time) and verify that Set Automatically is enabled. Failure to resolve could lead to serious under-delivery or over-delivery of insulin. If you did not change the time, contact Tidepool Support.", comment: "Time change alert body")
```

Comment:
This should interpolate using the bundle app name for app name, and the "contact Tidepool Support" needs to be turned into a non Tidepool specific version.  I would recommend just removing the whole sentence "If you did not change the time, contact Tidepool Support." We'll have to add it back with the Tidepool specific localization work.

Reply:
Yes, this is what I brought up in standup.  I am going to let this go until LOOP-1893 is addressed.

Reply:
The string that is stored here is not fixed by LOOP-1893.  LOOP-1893 allows us have Tidepool Loop display Tidepool specific strings, when non-Tidepool specific strings are in the code. Right now, pre LOOP-1893, the current state is that we need to store non-Tidepool specific strings, and display non-Tidepool specific strings, so we can continue to do DIY syncs. Does that make sense? The app name interpolation is also not blocked by LOOP-1893.